hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -182,13 +199,18 @@ public BackgroundTaskResult call() throws Exception {
             String snapshotBucketKey = dbBucketKey + OzoneConsts.OM_KEY_PREFIX;
             iterator.seek(snapshotBucketKey);
 
-            while (deletedIterator.hasNext()) {
+            int deletionCount = 0;
+            while (deletedIterator.hasNext() &&
+                deletionCount <= keyLimitPerSnapshot) {
```

Comment:
We could add a block comment right above the loop explaining that we are trying to gather as much key blocks as possible in one SCM block deletion request here.

Reply:
We can add it once we are sending the actual request to SCM